shared/config_reader/CMakeLists.txt

```diff
@@ -19,8 +19,12 @@ project(config_reader)
 
 cmake_minimum_required(VERSION 3.5)
 
-## Compile as C++14, supported in ROS Kinetic and newer
-add_compile_options(-std=c++14)
+# Default to C++14
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
I changed it to this because as I was facing a crazy bug, one of the things that I tried was to use an Ubuntu 22 + humble docker image as base, and you need this for it to compile fine in that env (otherwise it seemed that it used C++17).

Also the good news is that I tested the entire pipeline until rn on humble and everything works exactly the same using this branch.

Reply:
is there no way in catkin (or whatever it's called now) to define some flags over the entire build?

Reply:
https://github.com/nasa/astrobee/issues/636